### ISSUE TITLE
at-rule-* rules are incorrectly triggered by less `@import` statement

### ISSUE THREAD

**Issue by pvdlg** [stylelint/stylelint#3796]
> Clearly describe the bug

Related to #3767. Less `@import` statement are incorrectly identified as an at-rule.

> Which rule, if any, is the bug related to?

All `at-rule-*` rules.

> What CSS is needed to reproduce the bug?

```less
@import 'something';

.class {

}
```

> What stylelint configuration is needed to reproduce the bug?

Any config with an `at-rule-*` rule.

> Which version of stylelint are you using?

`9.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "**/*.less"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Less

> What did you expect to happen?

Not trigger `at-rule-*` rule for statement that are not an at-rule (e.g `@import`).

**Comment by jeddy3** [comment]
@pvdlg Thanks for the report and for using the template.

I believe the intended behaviour is that `@import` is considered an at-rule regardless of the parser. It was a long-standing bug that they weren't considered that when the Less parser was used. That was fixed in [`9.7.0`](https://github.com/stylelint/stylelint/blob/master/CHANGELOG.md#970). 

If you're not expecting this behaviour can you help us out by going into more detail, please?

**Comment by pvdlg** [comment]
Oh I didn't realize that....
The problem I have is with the following code:
```less
@import 'styles/syntax-variables.less';
@import 'styles/editor.less';
@import 'styles/base.less';

// Languages
@import 'styles/languages/css.less';
@import 'styles/languages/gfm.less';
@import 'styles/languages/javascript.less';
@import 'styles/languages/json.less';
@import 'styles/languages/less.less';
@import 'styles/languages/markdown.less';
@import 'styles/languages/sass.less';
@import 'styles/languages/scss.less';

// Plugins
@import 'styles/plugins/minimap.less';
```

With the config `at-rule-semicolon-newline-after: 'always'` it produces the errors:
```
  3:28  ✖  Expected newline after ";"   at-rule-semicolon-newline-after
 13:38  ✖  Expected newline after ";"   at-rule-semicolon-newline-after
```

That doesn't seems right as there is a newline after lines 3 and 13. It seems the rule return an error when you have 2 new lines. According to the rule documentation it's supposed to enforce "at least a new line" and not "strictly one new line"

So that seems to be more an issue with the `at-rule-semicolon-newline-after` and not with the detection of at-rules.

**Comment by pvdlg** [comment]
I just realize that the following code is valid:
```less
@import 'styles/syntax-variables.less';
@import 'styles/editor.less';
@import 'styles/base.less';

/* Languages */
@import 'styles/languages/css.less';
@import 'styles/languages/gfm.less';
@import 'styles/languages/javascript.less';
@import 'styles/languages/json.less';
@import 'styles/languages/less.less';
@import 'styles/languages/markdown.less';
@import 'styles/languages/sass.less';
@import 'styles/languages/scss.less';

/*  Plugins */
@import 'styles/plugins/minimap.less';
```

So maybe it's related to the comment detection in Less. It seems that `/* comment */ ` is considered a comment by stylelint but not `// comment`.
However both are valid in Less: http://lesscss.org/#comments

**Comment by pvdlg** [comment]
So I think my problem is #3754.
I'm gonna close this issue then. Sorry for the wrong report.

**Comment by jeddy3** [comment]
@pvdlg Thanks for the further explanation.

> So I think my problem is #3754.

Yep, I think you're right. There's a regression in our support for single-line Less comments that's causing side effects in various rules. Both `// comments` and `/* comments */` should work. Unfortunately, as I said in https://github.com/stylelint/stylelint/issues/3754#issuecomment-434490220, it will require a bit of digging to uncover if the problem is with stylelint or with the Less parser we use. We rely on contribution to support Less so that issue is waiting for a Less user to pick it up.

**Comment by xuhen** [comment]
> I just realize that the following code is valid:
> 
> ```css-less
> @import 'styles/syntax-variables.less';
> @import 'styles/editor.less';
> @import 'styles/base.less';
> 
> /* Languages */
> @import 'styles/languages/css.less';
> @import 'styles/languages/gfm.less';
> @import 'styles/languages/javascript.less';
> @import 'styles/languages/json.less';
> @import 'styles/languages/less.less';
> @import 'styles/languages/markdown.less';
> @import 'styles/languages/sass.less';
> @import 'styles/languages/scss.less';
> 
> /*  Plugins */
> @import 'styles/plugins/minimap.less';
> ```
> 
> So maybe it's related to the comment detection in Less. It seems that `/* comment */ ` is considered a comment by stylelint but not `// comment`.
> However both are valid in Less: http://lesscss.org/#comments

thx a lot, that's works for me
